leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/serialization/EndpointContextSerDes.java

```diff
@@ -121,28 +122,42 @@ public static EndpointContext deserialize(JsonObject peer) {
             Attributes attributes = new Attributes();
             for (Member member : value.asObject()) {
                 String name = member.getName();
-                String attributeValue = member.getValue().asString();
-                // convert the text values into typed values according their name
-                if (name.equals(UdpEndpointContext.KEY_PLAIN)) {
-                    attributes.add(name, attributeValue);
-                } else if (name.equals(DtlsEndpointContext.KEY_SESSION_ID)) {
-                    attributes.add(name, new SessionId(StringUtil.hex2ByteArray(attributeValue)));
-                } else if (name.equals(DtlsEndpointContext.KEY_EPOCH)) {
-                    attributes.add(name, Integer.parseInt(attributeValue));
-                } else if (name.equals(DtlsEndpointContext.KEY_CIPHER)) {
-                    attributes.add(name, attributeValue);
-                } else if (name.equals(DtlsEndpointContext.KEY_HANDSHAKE_TIMESTAMP)) {
-                    attributes.add(name, Long.parseLong(attributeValue));
-                } else if (name.equals(DtlsEndpointContext.KEY_READ_CONNECTION_ID)) {
-                    attributes.add(name, new ConnectionId(StringUtil.hex2ByteArray(attributeValue)));
-                } else if (name.equals(DtlsEndpointContext.KEY_WRITE_CONNECTION_ID)) {
-                    attributes.add(name, new ConnectionId(StringUtil.hex2ByteArray(attributeValue)));
+                Definition<?> key = MapBasedEndpointContext.ATTRIBUTE_DEFINITIONS.get(name);
+                if (key != null) {
```

Comment:
I feel if key == null we should at least LOG a WARN

Reply:
Done.